Sources/update-strings/Commands/SortCommand.swift

```diff
@@ -27,15 +30,35 @@ struct SortCommand: ParsableCommand {
         
         print("  Found \(stringsFileURLs.count)")
         
+        var unsortedStringPaths: [String] = []
+        
         for stringsFileURL in stringsFileURLs {
             let relativePath = stringsFileURL.path
                 .replacingOccurrences(of: currentDirectoryURL.path, with: "")
                 .trimmingCharacters(in: CharacterSet(charactersIn: "/"))
             
-            print("Sorting \(relativePath)")
+            let currentStrings = try String(contentsOf: stringsFileURL)
             
             let stringsFile = try StringsFile(url: stringsFileURL)
-            try stringsFile.description.write(to: stringsFileURL, atomically: true, encoding: .utf8)
+            let newStrings = stringsFile.description
+            
+            guard currentStrings != newStrings else { continue }
+            
+            print("\(dryRun ? "Validating" : "Sorting") \(relativePath)")
+            
+            if dryRun {
+                unsortedStringPaths.append(relativePath)
+            } else {
+                try stringsFile.description.write(to: stringsFileURL, atomically: true, encoding: .utf8)
+            }
+        }
+        
+        if dryRun, !unsortedStringPaths.isEmpty {
+            throw ValidationError(
+                unsortedStringPaths
+                    .map { "\($0) is not sorted." }
+                    .joined(separator: "\n")
+            )
```

Comment:
I didn't know `ValidationError` and some other error are provided by argument parser, that's good to know. I guess that's where the nicer formatting comes from. 

I don't think this should necessarily be shown as an error since that implies the command didn't finish running correctly. You could do this, the way the paths are a little nicer to read too since they're all intended the same (ie. not indented).
```suggestion
            print("The following files need to be sorted.")
            unsortedStringPaths.forEach { print($0) }
            
            throw ExitCode.failure
```